lib/floe/workflow/choice_rule.rb

```diff
@@ -42,8 +44,22 @@ def true?(*)
 
       private
 
-      def variable_value(context, input)
-        Path.value(variable, context, input)
+      def validate_next!(workflow)
+        if is_child?
+          # non-top level nodes don't allow a next
+          invalid_field_error!("Next", @next, "not allowed in a child rule") if @next
+        elsif !@next
+          # top level nodes require a next
+          missing_field_error!("Next")
+        elsif !workflow_state?(@next, workflow)
+          # top level nodes require a next field that is found
+          invalid_field_error!("Next", @next, "is not found in \"States\"")
+        end
+      end
+
+      # returns true if this is a child rule underneath an And/Or/Not
+      def is_child? # rubocop:disable Naming/PredicateName
+        !(%w[And Or Not] & name[0..-2]).empty?
```

Comment:
Is name an Array?  something about the variable "name" is mentally confusing cause it feels like it would be a String, but & can only work with an Array.

Reply:
Soemthing about this logic is so confusing for me (probably that's why I wrote include_any and include_none in more_core haha).  Can you give an example here what name would look like and how it would match or not match?

Reply:
I had `%w[And Or Not].include?(name[-3])` but this was confusing to Adam.

```json
{
  "Or": [
    {"Variable": "$.foo", "IsString": true},
    {"Variable": "$.foo", "IsBoolean": true}
  ], "Next": "Finished"
}
```

Basically, the paths will be:
- `[..., "Choices", "Or"]` -- require a "Next".
- `[..., "Choices", "Or", 1, "Data"]` -- no "Next" allowed.
- `[..., "Choices", "Or", 2, "Data"]` -- no "Next" allowed.

Note, the first one has an "Or", but it is not in the parent, hence the `name[0..-2]`

@agrare you still prefer the `& name[0..-2]` over `include?(name[-3])`?

Reply:
Yeah it doesn't read great and my suggestion probably made it worse, I was originally concerned that .include? would be doing a substring search and other names could accidentally match but actually that might not be possible if name is an array and not a flat string.

Reply:
We have specs for this so we can always come back and improve this as we go